src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy

```diff
@@ -29,9 +31,28 @@ class JavaDistributionPlugin implements Plugin<Project> {
 
         DistributionExtension ext = project.extensions.create('distribution', DistributionExtension)
 
+        // Specify classpath using pathing jar rather than command line argument on Windows, since
+        // Windows path sizes are limited.
+        PathingJarTask pathingJar = project.tasks.create("pathingJar", PathingJarTask)
+
         Task startScripts = project.tasks.create('createStartScripts', DistributionCreateStartScriptsTask, {
             description = "Generates standard Java start scripts."
-        })
+        }) << {
+            if (ext.isEnablePathingJar()) {
+                // Replace standard classpath with pathing jar in order to circumnavigate length limits:
+                // https://issues.gradle.org/browse/GRADLE-2992
+                def winScriptFile = project.file getWindowsScript()
+                def winFileText = winScriptFile.text
+
+                // Remove too-long-classpath and use pathing jar instead
+                winFileText = winFileText.replaceAll('set CLASSPATH=.*', 'rem CLASSPATH declaration removed.')
+                winFileText = winFileText.replaceAll(
+                    '("%JAVA_EXE%" .* -classpath ")%CLASSPATH%(" .*)',
+                    '$1%APP_HOME%\\\\lib\\\\' + pathingJar.archiveName + '$2')
+
+                winScriptFile.text = winFileText
```

Comment:
i'm tempted for us to bundle our own template batch file here as we can, as part of the CreateStartScripts task, override the default one Gradle uses.